### PR TITLE
feat: added a copy button to the sql bar

### PR DESCRIPTION
**Pull request by cooldude6000**
Closes: #10602 

### Description:

Hey,
So I added the copy sql button as shown in the below video. BUT, I am unable to get the the value of the sql (of what to copy). I tried importing it from `SqlRunner.txt` , but I was unable to because of some scope errors and also I dont know if its the right file to import from or not. So please, let me know from where I can import that sql.

<-- Even better add a screenshot / gif / loom -->

https://github.com/user-attachments/assets/3dc99271-e1d5-4a9d-9375-78eedc8f749b

### Reviewer actions

- [x] I have manually tested the changes in the preview environment
- [x] I have reviewed the code
- [x] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/common/CollapsableCard.tsx [hunk 1]
```diff
@@ -125,6 +134,30 @@ const CollapsableCard: FC<React.PropsWithChildren<CollapsableCardProps>> = ({
                     <Group spacing="xs">{headerElement}</Group>
                 </Group>
 
+                {!isOpen && title === 'SQL' && (
+                    <CopyButton value="{sql}" timeout={1000}>
```
[the remark this answers is not in the document]
Comment (cooldude6000) [reply packages/frontend/src/components/common/CollapsableCard.tsx:138]: Hey, I tried using {sql} , but when I copy the value was showing as **undefined**. So I kept it like "{sql}" just to show that the sql value/data should be present there; But I wasn't able to get the SQL data to copy (Idk from where I need to import the sql from ). So I was asking you where can I find and import that SQL value.

[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/components/common/CollapsableCard.tsx:138]: Ok so this won't copy the actual sql... This should be received as a prop or in some way through the context API...

**Comment by joaoviana** [comment]
Thanks for the contribution @cooldude6000, I will close this PR until you find a working solution. Please feel free to reopen a new one once you get it working 👍 

A side-note on the branch name: we should avoid adding special chars like quotation marks for the branch names. 

Looking forward to seeing a working solution !

File: packages/frontend/src/components/common/CollapsableCard.tsx [hunk 2]
```diff
@@ -61,6 +69,7 @@ const CollapsableCard: FC<React.PropsWithChildren<CollapsableCardProps>> = ({
     headerElement,
     rightHeaderElement,
     minHeight = 300,
+    // sql,
```
[the remark this answers is not in the document]
Comment (cooldude6000) [reply packages/frontend/src/components/common/CollapsableCard.tsx:72]: Yes. But idk from which component/file I need to pass the sql value to this `CollapsableCard.tsx` . So i just commented out that we need to get the sql value from whatever the file it is. So yes, according to my limited knowledge, ig it is necessary😅. Pls Lmk WDYT

**Comment by cooldude6000** [comment]
Can I DM you pls

**Comment by cooldude6000** [comment]
hey, i just submitted my NEW pr. Its WORKING. #10723 
Have a look at it!

**Pull request closed without merging**